expat/lib/xmlparse.c

```diff
@@ -1794,6 +1840,65 @@ XML_SetHashSalt(XML_Parser parser,
   return 1;
 }
 
+enum XML_Status XMLCALL
+XML_SetOption(XML_Parser parser,
+              enum XML_Option option,
+              void *value)
+{
+  if ((parser == NULL) || (value == NULL))
+    return XML_STATUS_ERROR;
+  /* block after XML_Parse()/XML_ParseBuffer() has been called */
+  if (parser->m_parsingStatus.parsing == XML_PARSING || parser->m_parsingStatus.parsing == XML_SUSPENDED)
+    return XML_STATUS_ERROR;
+  switch(option) {
+    case XML_OPTION_HUGE_XML:
+      parser->m_limit->hugeXML = *(XML_Bool*)(value) ? XML_TRUE : XML_FALSE;
+      return XML_STATUS_OK;
+    case XML_OPTION_NESTING_LIMIT:
+      parser->m_limit->nestingLimit = *(int *)(value);
+      return XML_STATUS_OK;
+    case XML_OPTION_EXPANSION_RATIO:
+      parser->m_limit->expansionRatio = *(int *)(value);
```

Comment:
We should reject values smaller than 2 here. 1 may work some of the time, ratio limit math needs it to be 2 or more.